tests/unit/backend/corpora/api_server/curator/collection/test_dataset.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from backend.corpora.common.corpora_orm import CollectionVisibility, UploadStatus
+from tests.unit.backend.corpora.api_server.base_api_test import BaseAuthAPITest
+
+
+class TestDeleteDataset(BaseAuthAPITest):
+    def test__delete_dataset(self):
+        curator_tag = "tag.h5ad"
+        processing_status = {"upload_status": UploadStatus.UPLOADING, "upload_progress": 10.0}
+        queries = ["dataset_uuid", "curator_tag"]
+        auth_credentials = [
+            (self.make_super_curator_header, "super", 202),
+            (self.get_auth_headers, "owner", 202),
+            (None, "none", 401),
+            (self.make_not_owner_header, "not_owner", 403),
+        ]
+        for query in queries:
+            for auth, auth_description, expected_status_code in auth_credentials:
```

Comment:
this is some fine test code modularity

Reply:
I was going to suggest we use multiple types of `UploadStatus` as well, but I guess that only matters if the sfn is actually processing. Speaking of which, our functional test suite(s) are lacking. 📉  Extra credit in Fit & Finish, perhaps...